src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3646,7 +3648,18 @@ GenTree* Lowering::LowerHWIntrinsicTernaryLogic(GenTreeHWIntrinsic* node)
                 if (condition->OperIsConvertMaskToVector())
                 {
                     GenTree* tmp = condition->AsHWIntrinsic()->Op(1);
-                    BlockRange().Remove(condition);
+
+                    LIR::Use use;
+                    if (BlockRange().TryGetUse(condition, &use))
+                    {
+                        use.ReplaceWith(tmp);
+                        BlockRange().Remove(condition);
```

Comment:
We're not removing `condition` in both paths because it should already be marked as "unused" and that should handle things, correct?